catalog/README.md

```diff
@@ -5,11 +5,87 @@ We want catalogs to access datasets used in the production of climate services.
 ## Attributes and controlled vocabulary
 In a catalog, each dataset is described by a set of attributes, forming a catalog entry. Based on consultations, it appears useful to include in those entries descriptive fields that go beyond simple dataset identification. That is, entries such as model resolution can be useful to filter datasets. 
 
-To ensure catalog clarity, it is important that attributes are described as much as possible by a well-defined controlled vocabulary (CV). A CV describes the attributes and the possible values it can take. Validating values ensures that search queries are not sabotaged by slight naming differences.   
+To ensure catalog clarity, it is important that attributes are described as much as possible by a well-defined controlled vocabulary (CV). A CV describes the attributes and the possible values it can take. Validating values ensures that search queries are not sabotaged by minor typos or naming differences.   
 
-CMIP defines a Data Reference Syntax (DRS) used to name files that is not identical to the CV, in part because the DRS syntax cannot include spaces, periods or parentheses. This is why there are often differences between the model or institute id found in the file name and that found in the global attributes metadata.    
 
-I suggest that we use the CV rather than the DRS to define the catalog and validate the entries. The DRS is still a useful guide as to the minimal subset of attributes uniquely identify an item in a CMIP collection. 
+## Implementation 
+
+### Data model
+
+Each catalog type (e.g. reanalysis) defines a _data model_, a representation of the attributes that make up a catalog entry. This data model is defined using pydantic, which makes it easy to impose rules and restrictions to the values entered in the catalog. It is possible, for example,  to constrain an attribute to a type, a list of possibilities or a regular expression. 
+
+The definition of CVs for attributes uses `pyessv`, the library underlying metadata validation in ES-DOC. The `pyessv-archive` repository is cloned locally and loaded by pyessv at import time. pyessv defines a naming hierarchy: authority, scope, collection, term. Within a data model, each attribute bound by a CV is typed as an `Enum`, with a list of _terms_ taken from a pyessv _collection_, for example:
```

Comment:
Is this cloning performed by pyessv or by the user? Is there a helper function for this?

Reply:
Fair enough. For my purposes, I added a git clone command to my `tox` configuration to better automate setup testing. Do you think the pyessv maintainers would appreciate a PR with helper function to do just that?